### PR TITLE
BREAKING: fix(Worker): Disable relative module specifiers

### LINKED ISSUE #5216: Worker: Disable relative import specifiers
A quick check on Chrome shows that relative import specifiers for Workers are resolved from `globalThis.location`. We don't have this concept at the moment so it should fail.

(Currently our `Worker` resolves from the main module, which *used to* be our `globalThis.location`.)

### PR DESCRIPTION
**Pull request by nayeemrmn**
Fixes #5216. @bartlomieju

### CODE REVIEW

File: cli/ops/worker_host.rs [hunk 1]
```diff
@@ -191,13 +191,11 @@ fn op_create_worker(
...
   let global_state = state.global_state.clone();
   let permissions = state.permissions.clone();
-  let referrer = state.main_module.to_string();
   let worker_id = state.next_worker_id;
   state.next_worker_id += 1;
   drop(state);
 
-  let module_specifier =
-    ModuleSpecifier::resolve_import(&specifier, &referrer)?;
+  let module_specifier = ModuleSpecifier::resolve_import(&specifier, "")?;
```
Comment (bartlomieju) [inline cli/ops/worker_host.rs:198]: use `ModuleSpecifier::resolve_url` and revert changes to `core/module_specifier.rs`?

File: docs/runtime/workers.md [hunk 2]
```diff
@@ -11,11 +11,11 @@ Currently Deno supports only `module` type workers; thus it's essential to pass
```
Comment (bartlomieju) [inline docs/runtime/workers.md:11]: Add note here about the `URL` business, preferable linking to the issue

**Comment by nayeemrmn** [comment]
@bartlomieju Please review. Based on [stability promises for web APIs](http://deno.antipy.cf/v1#stability), this can be minorly released?

**Comment by bartlomieju** [comment]
> Based on [stability promises for web APIs](http://deno.antipy.cf/v1#stability), this can be minorly released?

Yes, it can. I'll review in the coming days - just want to cool off cli changes for a few days :)

**Comment by wongjiahau** [comment]
@nayeemrmn will this approach be broken on Windows?

**Comment by nayeemrmn** [comment]
> @nayeemrmn will this approach be broken on Windows?

You mean how `new URL(...).pathname` is broken on Windows? No. The worker script takes a module specifier so we don't give it a path, we pass `new URL(...).href` as you can see.

**Pull request merged**